src/List.mo

```diff
@@ -25,10 +51,26 @@ module {
   };
 
   /// Construct a list by pre-pending a value.
-  /// This function is similar to a `list.cons(item)` function.
+  ///
+  /// Example:
+  /// ```motoko include=initialize
+  /// List.push<Nat>(0, null) // => ?(0, null);
+  /// ```
+  ///
+  /// Runtime: O(1)
+  ///
+  /// Space: O(1)
   public func push<T>(x : T, l : List<T>) : List<T> = ?(x, l);
 
   /// Return the last element of the list, if present.
+  /// Example:
+  /// ```motoko include=initialize
+  /// List.last<Nat>(?(0, ?(1, null))) // => ?1
```

Comment:
```suggestion
  /// var list = List.nil<Nat>();
  /// list := List.push<Nat>(1, list);
  /// list := List.push<Nat>(0, list);
  /// List.last<Nat>(list) // => ?1
```
I think it be better to try and use the API instead of the option notation when possible in the examples

Reply:
But then its harder to relate the intpreter output to the input>
```
List.rev(List.fromArray [1, 2, 3]) // => ?(3 (?2, ?(1, null))
````
or
``` 
List.rev(List.fromArray [1, 2, 3]) // =>List.fromArray [3, 2, 1]
````
but the user actually sees `?3 (?2 (?(1,0))` in the console.
I'm not sure thats actually better.


```
List.rev(?(1, ?(2, ?(3, null)) // rev '1,2,3'  
// => ?(3 (?2, ?(1, null)) // yields '3,2,1'
````